### PR TITLE
Various optimizations

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
I think most commits are best to review separately

Review/merge this one after #57

### CODE REVIEW

File: src/sensors/DepthCamera.cpp [hunk 1]
```diff
@@ -98,6 +98,20 @@ void DepthCamera::render(const RenderOptions& opt, RenderResult& res) const {
...
+        const cv::Point2d& p_2d = project3Dto2D(point);
+        if (0 <= p_2d.x <= res.getWidth() && 0 <= p_2d.y <= res.getHeight())
+        {
+            in_view = true;
+            break;
+        }
+    }
+    if (!in_view)
+        return;
+
```
Comment (PetervDooren) [inline src/sensors/DepthCamera.cpp]: A long wall might have its points far away, but the faces of the mesh should still be rendered

File: src/sensors/DepthCamera.cpp [hunk 2]
```diff
@@ -96,20 +96,19 @@ void DepthCamera::render(const RenderOptions& opt, RenderResult& res) const {
 
     // transform points
     std::vector<geo::Vec3d> points_t(points.size());
-
-    for(unsigned int i = 0; i < points.size(); ++i) {
-        points_t[i] = pose * points[i];
-    }
+    std::vector<bool> points_t_in(points.size());
```
Comment (PetervDooren) [inline src/sensors/DepthCamera.cpp]: Pick a better name, e.g. points_t_in_view.

File: include/geolib/sensors/DepthCamera.h [hunk 3]
```diff
@@ -249,6 +249,8 @@ class DepthCamera {
 
     inline double getOpticalTranslationY() const { return ty_; }
 
+    inline bool initialized() const { return cache_valid_; }
```
Comment (PetervDooren) [inline include/geolib/sensors/DepthCamera.h]: cache_ valid is niet hetzelfde als initialized.

**Review by PetervDooren — CHANGES_REQUESTED** [review]

File: include/geolib/sensors/DepthCamera.h (hunk 3 above)
Comment (MatthijsBurgh) [reply include/geolib/sensors/DepthCamera.h]: But it is a good representation. And useless to introduce a second boolean with the same behaviour.

Comment (PetervDooren) [reply include/geolib/sensors/DepthCamera.h]: Could you add a docstring to define exactly what initialized means in this context.

File: include/geolib/sensors/DepthCamera.h [hunk 4]
```diff
@@ -236,8 +249,12 @@ class DepthCamera {
...
```
Comment (PetervDooren) [inline include/geolib/sensors/DepthCamera.h]: Please add one or more docstrings to define what the behaviour of initialised() is. Does it mean the model is configured?

Comment (PetervDooren) [inline include/geolib/sensors/DepthCamera.h]: Please add one or more docstrings to define what the behaviour of initialised() is. Does it mean the model is configured?

**Review by PetervDooren — CHANGES_REQUESTED** [review]

**Review by PetervDooren — APPROVED** [review]

**Pull request merged**